### ISSUE TITLE
[Bug]: HuggingFace models keep getting downloaded

### ISSUE THREAD

**Issue by mederka** [speechbrain/speechbrain#1830]
### Describe the bug

We're working on a remote code execution system that would let users run functions on our infrastructure. The following function generates audio from an input text by using the Speechbrain Tacotron2 + HifiGan example:

```python
def tts(text: str):
    from speechbrain.pretrained import Tacotron2, HIFIGAN
    import torchaudio
    import io
    import os

    # Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
    tacotron2 = Tacotron2.from_hparams(
        source="speechbrain/tts-tacotron2-ljspeech",
        savedir="/data/tmpdir_tts")
    hifi_gan = HIFIGAN.from_hparams(
        source="speechbrain/tts-hifigan-ljspeech",
        savedir="/data/tmpdir_vocoder")

    # Running the TTS
    mel_output, mel_length, alignment = tacotron2.encode_text(text)

    # Running Vocoder (spectrogram-to-waveform)
    waveforms = hifi_gan.decode_batch(mel_output)

    buffer = io.BytesIO()
    torchaudio.save(buffer, waveforms.squeeze(1), 22050, format="wav")
    return buffer
```
where `/data` is a persistent directory dedicated for caching. Each time this function is called, the model artifacts are downloaded, instead of using the cached files. This is wasteful and slows down function execution.

### Expected behaviour

The model artifacts should be downloaded only once into a cache directory. Subsequent `from_hparams` calls should use cached model artifacts.

**Comment by Adel-Moumen** [comment]
Hi @anautsch, could you please have a look? Thanks! :-)

**Comment by Adel-Moumen** [comment]
Hi @mederka,

Thanks for raising this issue. 

Have you tried to put the exact path of the downloaded model ? e.g. : 

```python

PATH_TO_THE_LOCAL_MODEL = "/data/tmpdir_tts/

def tts(text: str):
    from speechbrain.pretrained import Tacotron2, HIFIGAN
    import torchaudio
    import io
    import os

    # Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
    tacotron2 = Tacotron2.from_hparams(
        source=PATH_TO_THE_LOCAL_MODEL+"/tacotron2/",
        savedir="/data/tmpdir_tts")
    hifi_gan = HIFIGAN.from_hparams(
        source=PATH_TO_THE_LOCAL_MODEL+"/hifi_gan/",
        savedir="/data/tmpdir_vocoder")

    # Running the TTS
    mel_output, mel_length, alignment = tacotron2.encode_text(text)

    # Running Vocoder (spectrogram-to-waveform)
    waveforms = hifi_gan.decode_batch(mel_output)

    buffer = io.BytesIO()
    torchaudio.save(buffer, waveforms.squeeze(1), 22050, format="wav")
    return buffer
``` 

As pointed here : https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/pretrained/fetching.py#L51 if the link is valid, then it will automatically create a symlink to the model and use it. 

So, it may be interesting for you to check if the model is at the exact location of where you are pointing it.

**Comment by mederka** [comment]
Hi @Adel-Moumen, I get this error when I try to user local dir as a source:
```
OSError: [Errno 40] Too many levels of symbolic links: '/data/tmpdir_tts/hyperparams.yaml'
```

**Comment by Adel-Moumen** [comment]
Hmmm weird. It does work on my end. 

I did this:
```python

# the relative or exact path of the model 
PATH_TO_THE_LOCAL_MODEL = "."


def tts(text: str):
    from speechbrain.pretrained import Tacotron2, HIFIGAN
    import torchaudio
    import io
    import os

    # check if the model is already downloaded
    if not os.path.exists(PATH_TO_THE_LOCAL_MODEL+"/tts-tacotron2-ljspeech/"):
    # Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
        tacotron2 = Tacotron2.from_hparams(
            source="speechbrain/tts-tacotron2-ljspeech",
            savedir=PATH_TO_THE_LOCAL_MODEL+"/tts-tacotron2-ljspeech/")

    if not os.path.exists(PATH_TO_THE_LOCAL_MODEL+"/tts-hifigan-ljspeech/"):
        hifi_gan = HIFIGAN.from_hparams(
            source="speechbrain/tts-hifigan-ljspeech",
            savedir=PATH_TO_THE_LOCAL_MODEL+"/tts-hifigan-ljspeech/")

    # Intialize TTS (tacotron2) and Vocoder (HiFIGAN)
    tacotron2 = Tacotron2.from_hparams(
        source=PATH_TO_THE_LOCAL_MODEL+"/tts-tacotron2-ljspeech/")
    hifi_gan = HIFIGAN.from_hparams(
        source=PATH_TO_THE_LOCAL_MODEL+"/tts-hifigan-ljspeech/")

    # Running the TTS
    mel_output, mel_length, alignment = tacotron2.encode_text(text)

    # Running Vocoder (spectrogram-to-waveform)
    waveforms = hifi_gan.decode_batch(mel_output)

    buffer = io.BytesIO()
    torchaudio.save(buffer, waveforms.squeeze(1), 22050, format="wav")
    return buffer

print(tts("hello world"))
```

**Comment by anautsch** [comment]
Hi @medrka there are two ways of handling that error: delete or debug.

Does it need to be under `/data` or would a local dir `data` be ok, too?
(Do you have writing rights for /data ... it's a `/` level entry.)

i.e. please try
```python
    tacotron2 = Tacotron2.from_hparams(
        source="speechbrain/tts-tacotron2-ljspeech",
        savedir="data/tmpdir_tts")
    hifi_gan = HIFIGAN.from_hparams(
        source="speechbrain/tts-hifigan-ljspeech",
        savedir="data/tmpdir_vocoder")
```
(it worked on my end - but local files)

**Comment by mederka** [comment]
Thank you, @Adel-Moumen and @anautsch! The updated code with path check worked! It would be nice if this check didn't have to be manual and the models wouldn't be redownloaded in `from_hparams` call.

@anautsch, I have to use `/data` because it's persisted across calls, where's the local environment gets destroyed.

Also, and this is not related to this issue, but if you have time, how do I get larger texts to process? Right now it seems like there's a limit on how big a text can be and when it's bigger than the limit I get non-legible short audio.

**Comment by anautsch** [comment]
cc @Gastron regarding #1817 

@mederka please write a bit more about "the local environment gets destroyed" - it seem related to the mentioned PR (but want to be sure I get your observation correctly).

@ycemsubakan do you have an idea for « how do I get larger texts to process? »

**Comment by mederka** [comment]
@anautsch, imagine running the `tts` function above inside a docker container that gets destroyed right after it's done executing the function. This docker container has a volume that is persistent and is mounted to `/data`.

**Comment by ycemsubakan** [comment]
@anautsch  @mederka

I think it's due to this line: 
https://github.com/speechbrain/speechbrain/blob/4b485a32f183e53d7807cbdaff0acc1e93952204/speechbrain/lobes/models/Tacotron2.py#L1216

It's basically limiting tacotron to output a mel spectrogram with 1000 time steps.

**Comment by Adel-Moumen** [comment]
Hello @mederka, 

Can we help with anything else? Please let us know if there's anything we can assist with. 
Thank you.

**Comment by anautsch** [comment]
Closing here.

@mederka as mentioned, we are working on it in another PR. The approach would then be: download all you need & explicitly define it is a local fetch. (cc @Gastron fyi)
